### DISCUSSION TITLE
Move Settings to Tab Bar

### DISCUSSION THREAD

**Discussion by chickdan** [jellyfin/Swiftfin#1269] [Discussions]
As someone who likes to fiddle with settings (and test things out to contribute to the code) I find it extra cumbersome to have to back out of whatever screen I am on to access the settings menu.

Additionally, with the introduction of the Admin Dashboard tools, users may find themselves trying to access the settings screen more often as well.

So I am proposing to move the Settings screen to the Tab Bar for quick and easy access. I already have the changes locally and it is far more convenient to access.

**Comment by LePips** [comment]
See: #1158

**Reply by chickdan** [reply]
ah, didn't see what was changed in that PR (I'm not a tvOS user so I haven't been keeping a close eye on those PRs).

> I had the thought to make the first tab, regardless of what it could ever be, to have the settings button. This would remove its strong relationship with Home.

This would still result in the same level of friction as any sub-screen of the first tab would hide the button. Plus after changing tabs you'd still have to go back to the first tab to get to settings. Unless I'm misunderstanding.

**Reply by LePips** [reply]
By that first draft of the rule, you are correct. However, could improve it with:

> If Settings is not a tab, then the first tab will have Settings in the toolbar.

We could even go further to allow settings to be on the last tab. So, anything is possible at the moment until we get to a more robust implementation.

**Reply by JPKribs** [reply]
As a note, I do have iOS implemented on that branch. That being said, iOS didn't feel very impactful since it was just the 3 tabs you could move around/hide. I implemented it, pulled it, then re-added it so that's why there are no screenshots.

I think that branch is probably way out of date now but I'm always interested in a second opinion! I mostly put this down because I thought this was the first step: https://github.com/jellyfin/Swiftfin/issues/1034.

**Reply by JPKribs** [reply]
Pardon the terrible screenshots but if the goal is quicker access to the admin dashboard, what if we mirrored something like this from Plex:

![image](https://github.com/user-attachments/assets/bd0a7e64-6476-42d0-be64-c1b04a9c4fc6)

If the user isAdministrator, they can enable that on their Home Screen. I was thinking this could appear to the left of the user icon for now? Then, if sessions are active & nowPlaying:

![image](https://github.com/user-attachments/assets/b68eb116-c75a-49d5-a989-8bad5cd46a75)

This is assuming the reason people would want to open the Admin Dashboard is to check in on who is online. If we have the icon on home, it gives that user digit at a glance. Then, clicking that can open directly to the Admin Dashboard. Probably polling the ActiveSessionsViewModel for the Home Screen onAppear then once every 60 seconds or so? I think we could just publish a nowPlaying in addition to the existing sessions from the view model then just filter it to nowPlaying != nil and then get a count of that set for this? That'd give a count of only the nowPlaying sessions to use for this icon.

We'd probably want to add a setting section for "Admin" that's only present if the user isAdministrator then include this "Admin Quickview" toggle or whatever better name someone comes up with.

If this sounds like a benefit, I'd be happy to work on it!


### Edit:

Here is a quick mock up of what I meant:

<img width="351" alt="Screenshot 2024-10-13 at 10 06 05" src="https://github.com/user-attachments/assets/0967b370-1110-4b0e-b50a-e6dd4f657c3c">

<img width="353" alt="Screenshot 2024-10-13 at 10 06 31" src="https://github.com/user-attachments/assets/ad597870-0ec4-497a-9582-449927ec3005">

<img width="351" alt="Screenshot 2024-10-13 at 10 10 36" src="https://github.com/user-attachments/assets/007e87c5-23c9-4446-bee7-129bd8a64b9b">

**Reply by chickdan** [reply]
@LePips I'm understanding the bigger picture now, I'll share some of my thoughts on the related topics.

@JPKribs the goal for me personally is quicker access to the Settings in general. However, I could definitely see there being a number of users who would benefit from _even quicker_ access to the Admin Dashboard. I like your mocks and I think it could be one of the dynamic options that have been discussed in #1034 and #1158
